crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -181,6 +190,16 @@ impl<Provider: ResolverProvider, InstalledPackages: InstalledPackagesProvider>
         provider: Provider,
         installed_packages: InstalledPackages,
     ) -> Result<Self, ResolveError> {
+        let requires_python = if markers.is_some() {
+            None
+        } else {
+            Some(
+                python_requirement
+                    .requires_python()
+                    .map(RequiresPython::to_marker_tree)
+                    .unwrap_or_else(|| MarkerTree::And(vec![])),
```

Comment:
Should this evaluate to `None` if `requires_python` is empty? What's the thinking behind `MarkerTree::And(vec![])`?

Reply:
(I sometimes wonder if we could get away with not having `Option<MarkerTree>` anywhere.)

Reply:
I guess I was hoping to avoid ever allowing `And(vec![])` or `Or(vec![])`. Ideally that would be impossible in the future...

Reply:
Yeah I had similar hopes for `regex` initially (e.g., banning `""` and `[^\s\S]`), and it just turns out that it's useful to have types like this be able to represent "always match" and "never match." For example, you _might_ be able to ban `And(vec![])`, but 1) you'll have to juggle it with `Option<MarkerTree>` which is annoying and 2) there an infinite number of `MarkerTree` values that have the same truth table as `And(vec![])`. Because of (2), you kinda need to deal with "always true" (and also, "never true") expressions anyway. So you might as well allow the trivial examples of them.